src/pages/OnboardingPersonalDetails/BaseOnboardingPersonalDetails.tsx

```diff
@@ -77,25 +110,29 @@ function BaseOnboardingPersonalDetails({currentUserPersonalDetails, shouldUseNat
         return errors;
     };
 
+    const handleGoBack = useCallback(() => {
+        Navigation.goBack();
+    }, []);
+
     const PersonalDetailsFooterInstance = <OfflineIndicator />;
 
     return (
         <View style={[styles.h100, styles.defaultModalContainer, shouldUseNativeStyles && styles.pt8]}>
             <HeaderWithBackButton
-                shouldShowBackButton={false}
-                iconFill={theme.iconColorfulBackground}
-                progressBarPercentage={33.3}
+                shouldShowBackButton
+                progressBarPercentage={OPEN_WORK_PAGE_PURPOSES.includes(onboardingPurposeSelected ?? '') ? 50 : 75}
+                onBackButtonPress={handleGoBack}
```

Comment:
Why do we need the useCallback here and we cannot just put the `Navigation.goBack()` here?